### PR TITLE
Clone strategy csi verify

### PR DESCRIPTION
**Pull request by brybacki**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by awels** [comment]
Is this still a draft?

**Comment by brybacki** [comment]
@awels  now it is ready for review

File: pkg/controller/datavolume-controller.go [hunk 1]
```diff
@@ -2035,6 +2071,7 @@ func (r *DatavolumeReconciler) emitFailureConditionEvent(dataVolume *cdiv1.DataV
 	}
 }
 
+// TODO:
```
Comment (awels) [inline pkg/controller/datavolume-controller.go:2074]: Leftover?

File: pkg/controller/datavolume-controller.go [hunk 2]
```diff
@@ -1390,6 +1405,27 @@ func (r *DatavolumeReconciler) reconcileProgressUpdate(datavolume *cdiv1.DataVol
 	return reconcile.Result{RequeueAfter: 2 * time.Second}, nil
 }
 
+func (r *DatavolumeReconciler) getCsiDriverForStorageClass(storageClassName *string) (bool, error) {
```
Comment (awels) [inline pkg/controller/datavolume-controller.go:1408]: I think this function should be named something like StorageClassCSIDriverExists or something of that nature. We are not actually returning the CSIDriver object, we are just determining if it exists, but the name suggests we are returning it.

File: tests/csiclone_test.go [hunk 3]
```diff
@@ -72,4 +72,35 @@ var _ = Describe("[vendor:cnv-qe@redhat.com][level:component][crit:high] CSI Vol
...
+		if f.IsCSIVolumeCloneStorageClassAvailable() {
+			Skip("Test should only run on non-csi storage")
+		}
+
+		By(fmt.Sprintf("configure storage profile %s", cloneStorageClassName))
+		utils.ConfigureCloneStrategy(f.CrClient, f.CdiClient, cloneStorageClassName, originalProfileSpec, cdiv1.CloneStrategyCsiClone)
+
+		dataVolume, _ := createDataVolumeDontWait("dv-csi-clone-test-1", utils.DefaultImagePath, v1.PersistentVolumeFilesystem, cloneStorageClassName, f)
+		waitForDvPhase(cdiv1.CloneScheduled, dataVolume, f)
+		verifyEvent(controller.ErrUnableToClone, dataVolume.Namespace, f)
```
Comment (awels) [inline tests/csiclone_test.go:85]: I think we would want to revert the clone strategy override after we are done with the test in an AfterEach maybe?

File: pkg/controller/datavolume-controller.go (hunk 2 above)
Comment (brybacki) [reply pkg/controller/datavolume-controller.go:1408]: Yes

File: tests/csiclone_test.go (hunk 3 above)
Comment (brybacki) [reply tests/csiclone_test.go:85]: This is actually what is done here:
https://github.com/kubevirt/containerized-data-importer/blob/80ca2de589616e6046c224471525242eba2d7a21/tests/csiclone_test.go#L35

```
AfterEach(func() {
		if originalProfileSpec != nil {
			By("Restore the profile")
			utils.UpdateStorageProfile(f.CrClient, cloneStorageClassName, *originalProfileSpec)
		}
	})
```

Comment (awels) [reply tests/csiclone_test.go:85]: Oh missed that wasn't part of this PR.

**Review by awels — APPROVED** [review]
/lgtm
/approve

**Pull request merged**